lib/onebox/engine/youtube_onebox.rb

```diff
@@ -13,11 +13,36 @@ class YoutubeOnebox
       WIDTH  ||= 480
       HEIGHT ||= 360
 
-      def placeholder_html
-        og = get_opengraph.data
+      def parse_embed_response
+        return unless video_id
+
```

Comment:
Memoization suggestion part 1/2:

```suggestion
        return @parse_embed_response if defined?(@parse_embed_response)
```